### DISCUSSION TITLE
Update Placeholder Component dynamically

### DISCUSSION THREAD

**Discussion by ucosatti** [filamentphp/filament#11250] [Help]
### Package

Form builder

### Package Version

3.2

### How can we help you?

how can I update dynamically a Placeholder Component content, with some data from the result of a query fired on afterUpdated from other InputField? (eg InputField loads an User ID, its fire a query and get its name, this name show be placed in the placeholder.)
Thanks.

**Comment by cirrion-software** [comment]
https://filamentphp.com/docs/3.x/forms/advanced#injecting-a-function-to-set-the-state-of-another-field

**Reply by ucosatti** [reply]
$set does not update the content of placeholder, it only updates states.

**Reply by cirrion-software** [reply]
https://filamentphp.com/docs/3.x/forms/layout/placeholder#dynamically-generating-placeholder-content

**Comment by leandrocfe** [comment]
Please share some code you are trying to do

**Comment by ucosatti** [comment]
After a while, I'd figure out how to update it. By injecting the Filament Component, get its container, then get the component by a key ( by name doesnt work).

```
TextInput::make('id')->required()->numeric()->live(debounce: 750)
->afterStateUpdated(
function ( ?string $state, FilamentComponet $component): void {
        if (!empty($state)) {
            $name = User::find($state)->get()->pluck("name");
            $component->getContainer()->getComponent("keyName")->content($name);
        }
    }
),
Placeholder::make("name")->content("")->key("keyName");
```
